src/serialize.h

```diff
@@ -7,7 +7,10 @@
 #define BITCOIN_SERIALIZE_H
 
 #include <attributes.h>
+#include <compat/assumptions.h> // IWYU pragma: keep
```

Comment:
The idea here is that this is basically our lowest-level header? Or because these assumptions mostly involve serialization?